internal/batches/executor/run_steps.go

```diff
@@ -119,8 +119,12 @@ func runSteps(ctx context.Context, opts *executionOpts) (result execution.Result
 			stepContext.Steps.Changes = previousStepResult.Files
 			stepContext.Outputs = opts.task.CachedResult.Outputs
 
-			if err := workspace.ApplyDiff(ctx, []byte(opts.task.CachedResult.Diff)); err != nil {
-				return execResult, nil, errors.Wrap(err, "getting changed files in step")
+			// If the previous steps made any modifications in the workspace yet,
+			// apply them.
+			if opts.task.CachedResult.Diff != "" {
```

Comment:
Are there existing tests to cover when `CachedResult.Diff` is blank and not blank?

Reply:
Another PR just found this bug, and it has a test for it so will merge this and then in the next PR you'll get that test :)